### ISSUE TITLE
Separate table for blog

### ISSUE THREAD

**Issue by s1-ranjan** [v17development/flarum-blog#84]
As of now, the blog plugin is using the discussions table to render blog post, it would be nice if blogs are been moved to a separate table

**Comment by jaspervriends** [comment]
Thank you for your feedback.

Due to the fact we use discussions as 'blog' posts, we can use the advantages of the following extensions: `Tags`, `Subscriptions`, `Lock`, `Flags`, `Likes`, `Mentions` and of course the comment sections of discussions.

It also makes the Blog extension very extensable as it can automatically implement features that were meant for discussions.

Would there be any reason why a seperate table would be prefered? Please let me know :)

**Comment by s1-ranjan** [comment]
While looking at the profiles page it's treated blogs and discussion as a discussion so it's would be good to go with a separate table, with this separate table this issue can be removed by making a separate subtab in the profile page for the blog, well regarding the extensions that are been meant for the discussion they can be extended too for the blog too.

**Comment by ctrought** [comment]
I don't think you need a separate table to do that, I am sure if the feature were desired they could ad a tab and show blog posts there, filtered separated from discussions. Like Jasper said, if it is a separate table this means they can't leverage existing extensions that use discussions. By moving to a separate table I imagine you are going to lose all this functionality.

> `Tags`, `Subscriptions`, `Lock`, `Flags`, `Likes`, `Mentions` and of course the comment sections of discussions.

I like your idea, but I don't think it is related to separating DB tables. Perhaps you can submit a separate feature request for if.

**Comment by ctml91** [comment]
I don't think you need a separate table to do that, I am sure if the feature were desired they could ad a tab and show blog posts there, filtered separated from discussions. Like Jasper said, if it is a separate table this means they can't leverage existing extensions that use discussions. By moving to a separate table I imagine you are going to lose all this functionality.

> Tags, Subscriptions, Lock, Flags, Likes, Mentions and of course the comment sections of discussions.

I like your idea, but I don't think it is related to separating DB tables. Perhaps you can submit a separate feature request for if.

**Comment by imorland** [comment]
Thanks for taking the time to report this.

This extension has since been transferred to FriendsOfFlarum and substantially reworked (new namespace, Flarum 1.8+/PHP 8.2+, updated dependencies). This issue is several years old and predates that work, so we believe it's either no longer reproducible or no longer relevant on the current release.

We're closing it to focus the tracker on issues confirmed against the current FoF version. If you're still hitting this on the latest release of `fof/blog`, please reopen (or open a new issue) with up-to-date reproduction steps and your `php flarum info` output — we'll gladly take another look.

Thanks again! 🙏
